### ISSUE TITLE
Exception when processing mapping from constants/expressions to inner fields with multi-arg mapping

### ISSUE THREAD

**Issue by Lysergid** [mapstruct/mapstruct#1552]
Mapping annotations processor throws NPE when trying to have multi-arg mapping method with inner field(s) mapped only from "constant" or "expression".

If inner field mapped not only from constant/expression but also from "source" mapping starts to be compilable. You can see it by uncommenting mapping statements.

Not really familiar with MapStruct code, but after brief investigation looks like issue with marking such fields as "processed" in BeanMappingMethod$Builder.handleDefinedMapping().

```
@Mapper
public interface FailingTwoArgsMapper {
    @Mapping(constant = "constant", target = "innerTargetOne.targetTwo")
//    @Mapping(source = "sourceOne", target = "innerTargetOne.targetOne")
    @Mapping(source = "sourceTwo", target = "innerTargetTwo.targetTwo")
    public Target twoArgsWithConstantFailed(String sourceOne, String sourceTwo);

    @Mapping(expression = "java(\"constant\")", target = "innerTargetOne.targetTwo")
//    @Mapping(source = "sourceOne", target = "innerTargetOne.targetOne")
    @Mapping(source = "sourceTwo", target = "innerTargetTwo.targetTwo")
    public Target twoArgsWithExpressionFailed(String sourceOne, String sourceTwo);

    @Mapping(constant = "constant", target = "innerTargetOne.targetTwo")
    @Mapping(source = "sourceOne", target = "innerTargetOne.targetOne")
    @Mapping(source = "sourceTwo", target = "innerTargetTwo.targetTwo")
    public Target twoArgsOk(String sourceOne, String sourceTwo);

    @Mapping(constant = "constant", target = "innerTargetOne.targetTwo")
    public Target oneArgOk(String sourceOne);

    static class Target {

        private InnerTarget innerTargetOne;
        private InnerTarget innerTargetTwo;

        public InnerTarget getInnerTargetOne() {
            return innerTargetOne;
        }

        public void setInnerTargetOne(InnerTarget innerTargetOne) {
            this.innerTargetOne = innerTargetOne;
        }

        public InnerTarget getInnerTargetTwo() {
            return innerTargetTwo;
        }

        public void setInnerTargetTwo(InnerTarget innerTargetTwo) {
            this.innerTargetTwo = innerTargetTwo;
        }
    }

    static class InnerTarget {

        private String targetOne;
        private String targetTwo;

        public String getTargetOne() {
            return targetOne;
        }

        public void setTargetOne(String targetOne) {
            this.targetOne = targetOne;
        }

        public String getTargetTwo() {
            return targetTwo;
        }

        public void setTargetTwo(String targetTwo) {
            this.targetTwo = targetTwo;
        }
    }
```

Exception thrown during compilation(annotation processing):
```
[8,8] Internal error in the mapping processor: java.lang.NullPointerException  
at org.mapstruct.ap.internal.util.Executables.isSetterMethod(Executables.java:112)  
at org.mapstruct.ap.internal.model.PropertyMapping$TargetWriteAccessorType.of(PropertyMapping.java:89)  
at org.mapstruct.ap.internal.model.PropertyMapping$MappingBuilderBase.targetWriteAccessor(PropertyMapping.java:143)  
at org.mapstruct.ap.internal.model.BeanMappingMethod$Builder.handleUnprocessedDefinedTargets(BeanMappingMethod.java:271)  
at org.mapstruct.ap.internal.model.BeanMappingMethod$Builder.build(BeanMappingMethod.java:152)  
at org.mapstruct.ap.internal.processor.MapperCreationProcessor.getMappingMethods(MapperCreationProcessor.java:373)  
at org.mapstruct.ap.internal.processor.MapperCreationProcessor.getMapper(MapperCreationProcessor.java:152)  
at org.mapstruct.ap.internal.processor.MapperCreationProcessor.process(MapperCreationProcessor.java:123)  
at org.mapstruct.ap.internal.processor.MapperCreationProcessor.process(MapperCreationProcessor.java:80)  
at org.mapstruct.ap.MappingProcessor.process(MappingProcessor.java:280)  
at org.mapstruct.ap.MappingProcessor.processMapperTypeElement(MappingProcessor.java:260)  
at org.mapstruct.ap.MappingProcessor.processMapperElements(MappingProcessor.java:226)  
at org.mapstruct.ap.MappingProcessor.process(MappingProcessor.java:162)  
at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:794)  
at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:705)  
at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1800(JavacProcessingEnvironment.java:91)  
at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1035)  
at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1176)  
at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1170)  
at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:856)  
at com.sun.tools.javac.main.Main.compile(Main.java:523)  
at com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:129)  
at com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:138)  
at org.codehaus.plexus.compiler.javac.JavaxToolsCompiler.compileInProcess(JavaxToolsCompiler.java:125)  
at org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile(JavacCompiler.java:171)  
at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:886)  
at org.apache.maven.plugin.compiler.CompilerMojo.execute(CompilerMojo.java:129)  
at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:132)  
at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)  
at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)  
at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)  
at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)  
at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)  
at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)  
at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:120)  
at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:347)  
at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:154)  
at org.apache.maven.cli.MavenCli.execute(MavenCli.java:582)  
at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:214)  
at org.apache.maven.cli.MavenCli.main(MavenCli.java:158)  
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)  
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)  
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)  
at java.lang.reflect.Method.invoke(Method.java:497)  
at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)  
at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)  
at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)  
at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)  
at org.codehaus.classworlds.Launcher.main(Launcher.java:46)  
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)  
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)  
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)  
at java.lang.reflect.Method.invoke(Method.java:497)  
at com.intellij.rt.execution.application.AppMain.main(AppMain.java:134)
```

**Comment by filiphr** [comment]
@Lysergid which version of MapStruct are you using?

**Comment by tinki007** [comment]
+1. Using single-source mapping with constant works fine and compiles. Adding second source to argument list throws already mentioned exception

**Comment by filiphr** [comment]
Thanks for the issue. I think I have found the problem and the fix is in PR #1585
